### PR TITLE
refactor(tests): migrate remaining tests ts

### PR DESCRIPTION
**Pull request by martinkaintas**

### CODE REVIEW

File: test/integration/accounts.ts
Comment (subhod-i) [inline test/integration/accounts.ts:34]: @davidyuk Do you think we need a global type for the SDK? we may have better clarity once stampit is removed.
@Kalovelo

File: test/integration/ga.ts
Comment (subhod-i) [inline test/integration/ga.ts:47]: You can reuse these types in the code while removing stamps in contracts section

Comment (subhod-i) [reply test/integration/ga.ts:47]: I mean, I see the same type of definitions in multiple places. Maybe we get to define them while removing stamps.  Not sure if we want to merge this now. we can merge this PR at last.

Comment (martinkaintas) [reply test/integration/ga.ts:47]: Absolutely, these types are declared in the [PR ](https://github.com/aeternity/aepp-sdk-js/blob/7d19119e44fc9c2e135238d2775996ef2e1ee234/src/contract/aci.ts)regarding the RPC module. Once that's merged I will remove them from here and update the PR. Should I rename this to WIP in this case?

Comment (subhod-i) [reply test/integration/ga.ts:47]: I am resolving this. A simple to-do comment will solve this.

File: test/integration/accounts.ts
Comment (subhod-i) [inline test/integration/accounts.ts:34]: Todo: replace any with type once all stamps are removed and we have a global class/type

File: test/integration/contract.ts
Comment (subhod-i) [inline test/integration/contract.ts:28]: todo: use definitions from the src

**Review by subhod-i — APPROVED** [review]
LGTM apart from the pending todos comments.

File: tsconfig.json
Comment (davidyuk) [inline tsconfig.json:25]: This make ts to add definitions of tests files to the build folder, we need another approach

File: tsconfig.json
Comment (davidyuk) [inline tsconfig.json:23]: avoid changing just formatting

File: test/utils.ts
Comment (davidyuk) [inline test/utils.ts:11]:
```suggestion
export function randomName (length: number): `${string}.chain` {
```

File: test/unit/memory-account.ts
Comment (davidyuk) [inline test/unit/memory-account.ts:47]: `return` is not necessary then (also change in other places)
```suggestion
    await acc.address().should.eventually.be.equal(testAcc.publicKey)
```

File: test/unit/memory-account.ts
Comment (davidyuk) [inline test/unit/memory-account.ts:46]:
```suggestion
    const acc = MemoryAccount({ keypair: testAcc })
```
didn't we already defined `MemoryAccount` as `_AccountMemory`?

File: test/unit/memory-account.ts
Comment (davidyuk) [inline test/unit/memory-account.ts:54]: fix https://github.com/aeternity/aepp-sdk-js/issues/1526 instead

File: test/unit/amount-formatter.ts
Comment (davidyuk) [inline test/unit/amount-formatter.ts:33]: `as const` works better in this case (should be already in develop)

File: test/integration/paying-for.ts
Comment (davidyuk) [inline test/integration/paying-for.ts:164]: why `42n` is not working here?

File: test/integration/ga.ts
Comment (davidyuk) [inline test/integration/ga.ts:148]: isn't it already a string?

File: test/integration/contract.ts
Comment (davidyuk) [inline test/integration/contract.ts:528]:
```suggestion
      oracleExtended.ttl.should.be.equal(+oracle.ttl + 50)
```
unary plus is more fragile than `parseInt`, it suits better for tests

File: test/integration/contract.ts
Comment (davidyuk) [inline test/integration/contract.ts:506]: these arguments shouldn't be required

File: test/integration/contract.ts
Comment (davidyuk) [inline test/integration/contract.ts:507]:
```suggestion
      contractId = contract.deployInfo.address ?? throw new Error()
```
🤔

File: package.json
Comment (davidyuk) [inline package.json:20]: `ts-node/register` already specified in `.mocharc.js`, also I specified accepted extensions there, so `/*` also not needed

**Review by davidyuk — CHANGES_REQUESTED** [review]

File: test/integration/contract-aci.ts
Comment (davidyuk) [inline test/integration/contract-aci.ts:314]: I found why are you doing this, it shouldn't be necessary after #1541

Comment (davidyuk) [inline test/integration/contract-aci.ts:314]: I found why are you doing this, it shouldn't be necessary after #1541

File: test/integration/ga.ts
Comment (martinkaintas) [reply test/integration/ga.ts:148]: It's `string | Buffer`

Comment (davidyuk) [reply test/integration/ga.ts:148]: `generateKeyPair` definition is fixed now

File: test/integration/rpc.ts
Comment (davidyuk) [inline test/integration/rpc.ts:175]:
```suggestion
        connected: { [keypair.publicKey]: {} },
```

File: test/integration/rpc.ts
Comment (davidyuk) [inline test/integration/rpc.ts:573]: just fix the return type of `WindowPostMessageFake` instead

**Review by davidyuk — CHANGES_REQUESTED** [review]

File: test/integration/contract.ts
Comment (martinkaintas) [reply test/integration/contract.ts:510]: We can add a function like this one, and use it to assert that `deployInfo.address` will be present at a certain scope. But I don't think we can make `deploy` itself do both.

```typescript
 export function assertAddress (address: any): asserts address is EncodedData<'ct'> {
 if (address == null) throw new IllegalArgumentError('Address is required')
}
```

**Review by davidyuk — APPROVED** [review]

**Pull request merged**
